### PR TITLE
Baseline data values: include all baselines across dates

### PR DESCRIPTION
**Pull request by emullaney**
~~Select baseline data based on least recent answer per contact.~~
Select baseline data points across all dates.
Update tests for this functionality.

### CODE REVIEW

File: tracpro/baseline/models.py [hunk 1]
```diff
@@ -32,7 +32,7 @@ class BaselineTerm(models.Model):
         chained_model_field='poll',
         auto_choose=True,
         related_name="baseline_terms",
-        help_text=_("The most recent response per user will be used as the baseline.")
+        help_text=_("The least recent response per user will be used as the baseline.")
```
Comment (rebecca-caktus) [inline tracpro/baseline/models.py]: Hmm, I don't think this is as clear in the reverse. What do you think about: "The first response in the time period per user..."

**Comment by rebecca-caktus** [comment]
Love how simple this change was. 

One silly `flake8` error, one comment, then good to :goat:

**Comment by emullaney** [comment]
This one is going to be a bit more complex to allow for multiple baselines over time. Going to re-think the wording and re-work the logic on this branch.

File: tracpro/baseline/forms.py [hunk 2]
```diff
@@ -68,7 +68,9 @@ class SpoofDataForm(forms.Form):
     start_date = forms.DateField(help_text=_(
         "Baseline poll data will be submitted on this date. " +
         "Follow up data will start on this date."))
-    end_date = forms.DateField(help_text=_("Follow up data will end on this date."))
+    end_date = forms.DateField(help_text=_(
+        "Follow up data will end on this date. " +
```
Comment (rebecca-caktus) [inline tracpro/baseline/forms.py]: `+` signs are unnecessary to concatenate strings across lines when in `()`.

File: tracpro/baseline/models.py [hunk 3]
```diff
@@ -33,7 +33,7 @@ class BaselineTerm(models.Model):
...
```
Comment (rebecca-caktus) [inline tracpro/baseline/models.py]: I think this help text needs to be updated.

File: tracpro/baseline/views.py [hunk 4]
```diff
@@ -144,8 +155,27 @@ def form_valid(self, form):
...
+            end = self.form.cleaned_data['end_date']
+
+            # Create a single PollRun for the Baseline Poll
+            self.create_baseline(baseline_question.poll, start, contacts, baseline_question,
+                                 baseline_minimum, baseline_maximum)
+
             # Create a PollRun for each date from start to end dates for the Follow Up Poll
+            loop_count = 1
             for follow_up_date in rrule.rrule(rrule.DAILY, dtstart=start, until=end):
+                if loop_count == 7:  # On the seventh day, add another set of baseline data
```
Comment (rebecca-caktus) [inline tracpro/baseline/views.py]: Hmm, if my finger-counting is correct I think this is one-off.

A possible tidy solution: 

```
for loop_count, follow_up_date in enumerate(rrule.rrule(rrule.DAILY, dtstart=start, until=end)):
    if loop_count % 7 == 0:
        ...
```

File: tracpro/baseline/views.py [hunk 5]
```diff
@@ -74,18 +74,40 @@ def get_context_data(self, **kwargs):
 
             region = self.request.region
 
-            baseline_dict = self.object.get_baseline(region=region)
-            context['baseline_dict'] = baseline_dict
+            baseline_dict, baseline_dates = self.object.get_baseline(region=region)
 
             follow_ups, dates = self.object.get_follow_up(region=region)
 
+            """
```
Comment (rebecca-caktus) [inline tracpro/baseline/views.py]: I'm not sure that I have ever seen block comments anywhere except for a docstring, but I don't have any good sources telling me that's the way it should be. Do you have a reason or reference for doing this?

Comment (emullaney) [inline tracpro/baseline/views.py]: If that's not the norm in python, I will go back to single # comments. No worries.

File: tracpro/baseline/views.py (hunk 4 above)
Comment (emullaney) [inline tracpro/baseline/views.py]: Cool stuff. Will re-do this. Forgot about enumerate().

**Comment by rebecca-caktus** [comment]
I see these warnings fly by when I create spoofed data:

```
RuntimeWarning: DateTimeField Response.updated_on received a naive datetime (2016-01-01 14:27:27.803316) while time zone support is active.
```

File: tracpro/baseline/views.py [hunk 6]
```diff
@@ -144,9 +150,27 @@ def form_valid(self, form):
...
+            contacts = self.form.cleaned_data['contacts']
+            baseline_minimum = self.form.cleaned_data['baseline_minimum']
+            baseline_maximum = self.form.cleaned_data['baseline_maximum']
+            follow_up_minimum = self.form.cleaned_data['follow_up_minimum']
+            follow_up_maximum = self.form.cleaned_data['follow_up_maximum']
+            start = self.form.cleaned_data['start_date']
+            end = self.form.cleaned_data['end_date']
+
+            # Create a single PollRun for the Baseline Poll
+            self.create_baseline(baseline_question.poll, start, contacts, baseline_question,
```
Comment (rebecca-caktus) [inline tracpro/baseline/views.py]: It looks like this could be moved into the for loop if you remove the `loop_count > 0` conditional.

Comment (emullaney) [inline tracpro/baseline/views.py]: Goooood point....

**Comment by rebecca-caktus** [comment]
:+1:

**Pull request merged**